streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -220,6 +223,47 @@ public String queryableStoreName() {
         return doMapValues(mapper, new MaterializedInternal<>(materialized, builder, MAPVALUES_NAME));
     }
 
+    @Override
+    public <VR> KTable<K, VR> transformValues(ValueTransformerWithKeySupplier<? super K, ? super V, ? extends VR> transformerSupplier, String... stateStoreNames) {
+        return doTransformValues(transformerSupplier, null, stateStoreNames);
+    }
+
+    @Override
+    public <VR> KTable<K, VR> transformValues(final ValueTransformerWithKeySupplier<? super K, ? super V, ? extends VR> transformerSupplier,
+                                              final Materialized<K, VR, KeyValueStore<Bytes, byte[]>> materialized,
+                                              final String... stateStoreNames) {
+        Objects.requireNonNull(materialized, "materialized can't be null");
+        return doTransformValues(transformerSupplier, materialized, stateStoreNames);
+    }
+
+    private <VR> KTable<K, VR> doTransformValues(final ValueTransformerWithKeySupplier<? super K, ? super V, ? extends VR> transformerSupplier,
+                                               final Materialized<K, VR, KeyValueStore<Bytes, byte[]>> materialized,
+                                               final String... stateStoreNames) {
+        Objects.requireNonNull(stateStoreNames, "stateStoreNames");
```

Comment:
Does this do the trick? The type is `String[]` and can only be `null` if somebody calls via
```
String[] stores = null;
table.transformValues(...., stores);
table.transformValues(...., null); // this would not trigger the check
```

Should better step through the array and check each entry for `null` ? Or is this intended and we rely on `connectProcessorAndStateStores` to do this check?

Reply:
Ah I now remembered: to guard against `(... , null)` we should follow the way in `InternalTopologyBuilder#addStateStore`. i.e.:

```
      if (stateStoreNames != null) {
            for (final String stateStoreName : stateStoreNames) {
                Objects.requireNonNull(processorName, "store name must not be null");
            }
        }
```